Sources/Mantis/CropViewController/CropViewController.swift

```diff
@@ -378,9 +420,42 @@ private extension CropViewController {
 extension CropViewController: CropViewDelegate {    
     func cropViewDidBecomeResettable(_ cropView: CropViewProtocol) {
         cropToolbar.handleCropViewDidBecomeResettable()
-        delegate?.cropViewControllerDidImageTransformed(self)
         delegate?.cropViewControllerDidImageTransformed(self, transformation: cropView.makeTransformation())
         delegate?.cropViewController(self, didBecomeResettable: true)
+        
+        if config.enableUndoRedo {
+            guard let previous = previousCropState else { return }
+            let userGenerated = isCropStateUserGenerated
+            currentCropState = cropView.makeCropState()
+            
+            if previous != currentCropState {
+                pushTransformRecordOntoStack(transformType: .transform, previous: previous, current: currentCropState, userGenerated: userGenerated)
+            }
+            
+            previousCropState = nil
+            currentCropState = nil
+        }
+    }
+    
+    private func pushTransformRecordOntoStack(transformType: TransformType, previous: CropState, current: CropState, userGenerated: Bool) {
```

Comment:
I think this function can be moved into TransformStack since it has nothing to do with CropViewController and more related to TransformStack.

Reply:
change made